recipes/openassetio/all/conanfile.py

```diff
@@ -0,0 +1,165 @@
+            self.requires("openssl/1.1.1s")
+            # TODO: cpython requires ncurses/6.2 but no pre-built package exists.
+            self.requires(f"ncurses/6.3")
+            self.requires(f"cpython/{self.options.python_version}")
+            self.requires("pybind11/2.10.1")
+
+    def validate(self):
+        if is_apple_os(self):
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support MacOS at this time"
+            )
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+
+        if is_msvc(self) and not self.dependencies["cpython"].options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} requires cpython:shared=True when using MSVC compiler")
+
+        check_min_vs(self, 191)
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def build_requirements(self):
+        self.tool_requires("cmake/3.25.1")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+
+        tc.variables["OPENASSETIO_ENABLE_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+
+        tc.variables["OPENASSETIO_GLIBCXX_USE_CXX11_ABI"] = self.settings.get_safe("compiler.libcxx") == "libstdc++11"
+
+        tc.variables["OPENASSETIO_ENABLE_PYTHON"] = self.options.with_python
+        if self.options.with_python:
+            tc.variables["Python_EXECUTABLE"] = self._python_exe
+            if is_msvc(self):
+                tc.variables["Python_LIBRARY"] = self._python_windows_lib
+
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    @property
+    def _python_exe(self):
+        # TODO: update to V2 once cpython is updated
+        return pathlib.Path(self.deps_user_info["cpython"].python).as_posix()
+
+    @property
+    def _python_windows_lib(self):
+        pth = pathlib.Path(
+            self.dependencies["cpython"].package_folder,
+            self.dependencies["cpython"].cpp_info.components["embed"].libdirs[0],
+            self.dependencies["cpython"].cpp_info.components["embed"].libs[0])
+        pth = pth.with_suffix(".lib")
+        return pth.as_posix()
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package_id(self):
+        if self.options.with_python:
+            self.info.requires["cpython"].minor_mode()
```

Comment:
Yeah this make me suspicious about the option for python version 🤔

Reply:
True it may not be needed if we also have an option for Python version, but I thought good practice regardless, and perhaps still needed for cases where `cpython` version is overridden by the consumer.

Reply:
This is exactly why it was concerning.

If you override the requirement from the consumer then the option would not actually do anything, the 1.x flow of when options are evaluated version the graph is not super obvious so I prefer to air on th side of caution.

Let the graph do graph stuff but not the options that package ABI which is a different topic